GUI/Types/Viewers/Resource.cs

```diff
@@ -394,46 +394,56 @@ public TabPage Create(VrfGuiContext vrfGuiContext, byte[] input)
-                    var tabVmat = new TabPage("vmat");
-                    tabVmat.Controls.Add(control);
-                    resTabs.TabPages.Add(tabVmat);
+                    var tab = new TabPage(name);
+                    tab.Controls.Add(control);
+                    resTabs.TabPages.Add(tab);
                 }
 
-                if (block.Type == BlockType.DATA && resource.ResourceType == ResourceType.EntityLump)
+                if (block.Type != BlockType.DATA)
                 {
-                    var control = new TextBox();
-                    control.Font = new Font(FontFamily.GenericMonospace, control.Font.Size);
-                    control.Text = Utils.Utils.NormalizeLineEndings(((EntityLump)block).ToEntityDumpString());
-                    control.Dock = DockStyle.Fill;
-                    control.Multiline = true;
-                    control.ReadOnly = true;
-                    control.ScrollBars = ScrollBars.Both;
-                    var tabEntities = new TabPage("Entities");
-                    tabEntities.Controls.Add(control);
-                    resTabs.TabPages.Add(tabEntities);
+                    continue;
                 }
 
-                if (block.Type == BlockType.DATA && resource.ResourceType == ResourceType.PostProcessing)
+                switch (resource.ResourceType)
                 {
-                    var control = new TextBox();
-                    control.Font = new Font(FontFamily.GenericMonospace, control.Font.Size);
-                    control.Text = Utils.Utils.NormalizeLineEndings(((PostProcessing)block).ToValvePostProcessing());
-                    control.Dock = DockStyle.Fill;
-                    control.Multiline = true;
-                    control.ReadOnly = true;
-                    control.ScrollBars = ScrollBars.Both;
-                    var tabVpost = new TabPage("vpost");
-                    tabVpost.Controls.Add(control);
-                    resTabs.TabPages.Add(tabVpost);
+                    case ResourceType.Material:
+                        AddContentTab(resTabs, "Reconstructed vmat", ((Material)block).ToValveMaterial());
+                        break;
+
+                    case ResourceType.EntityLump:
+                        AddContentTab(resTabs, "Entities", ((EntityLump)block).ToEntityDumpString());
+                        break;
+
+                    case ResourceType.PostProcessing:
+                        AddContentTab(resTabs, "Reconstructed vpost", ((PostProcessing)block).ToValvePostProcessing());
+                        break;
+
+                    case ResourceType.Texture:
+                        {
+                            if (FileExtract.IsChildResource(resource))
+                            {
+                                break;
+                            }
+
+                            var textureExtract = new TextureExtract(resource.FileName, (Texture)resource.DataBlock);
```

Comment:
this will call GenerateBitmap, which is slow and you don't need here.

Reply:
Fixed, bitmap is only generated when forming the content file for export.